File: CPP/203_RemoveLinkedListElements.cpp
// 230 - Remove Linked List Elements

ListNode* removeElements(ListNode* head, int val) {
    if(head == nullptr) return nullptr;

    ListNode* node = new ListNode(0);
    node->next = head;
    ListNode* curr = node;
    
    while(curr->next != nullptr) {
        if(curr->next->val == val) {
            curr->next = curr->next->next;
            continue;
        }
        curr = curr->next;
    }

    return node->next;
}

File: Java/203_RemoveLinkedListElements.java
// 203 - Remove Linked List Elements

public ListNode removeElements(ListNode head, int val) {
    if(head == null) return null;

    ListNode node = new ListNode(0);
    node.next = head;
    ListNode curr = node;

    while(curr.next != null) {
        if(curr.next.val == val) {
            curr.next = curr.next.next;
            continue;
        }
        curr = curr.next;
    }

    return node.next;
}